libtenzir/builtins/operators/unroll.cpp

```diff
@@ -204,13 +239,21 @@ class unroll_operator final : public crtp_operator<unroll_operator> {
           return resolve(field, slice.schema())
             .match(
               [&](offset result) -> std::optional<offset> {
+                if (result.empty()) {
+                  diagnostic::error("cannot unroll `this`")
+                    .primary(field)
+                    .emit(ctrl.diagnostics());
+                  return {};
+                }
```

Comment:
Is this still true?

Reply:
Also, this could be checked for earlier.

Reply:
It's just not needed anymore.

Reply:
I made a few adjustments that make this work now.